arm_robots/src/arm_robots/robot.py

```diff
@@ -342,10 +407,21 @@ def follow_jacobian_to_position(self,
     def get_both_arm_joints(self):
         return self.get_left_arm_joints() + self.get_right_arm_joints()
 
-    def get_joint_names(self, group_name: str = 'whole_body'):
-        return self.get_move_group_commander(group_name).get_active_joints()
+    def get_joint_names(self, group_name: Optional[str] = None):
+        # NOTE: Getting a list of joint names should not require anything besides a lookup on the ros parameter server.
+        #   However, MoveIt does not have this implemented that way, it requires connecting to the move group node.
+        #   This causes problems sometimes, so to simplify things we do the lookup manually if no group name is given.
+        if group_name is None:
+            robot = urdf.from_parameter_server()
+            active_joint_names = [j.name for j in robot.joints if j.type != 'fixed']
+            return active_joint_names
+        else:
+            if hasattr(self.robot_commander, 'get_active_joint_names'):
+                return self.robot_commander.get_active_joint_names(group_name)
+            else:
+                return self.get_move_group_commander('whole_body').get_active_joints()
```

Comment:
As currently written, in this last case `group_name` is silently ignored. Pretty confusing. I'd rather this throw an error than return the wrong set of joints

Reply:
oh I think that's just a mistake, it should  use group_name